File: app/src/main/java/com/lany/itemdecoration/sample/MainActivity.java
package com.lany.itemdecoration.sample;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button:
                startActivity(new Intent(this, LinearLayoutManagerActivity.class));
                break;
            case R.id.button2:
                startActivity(new Intent(this, GridLayoutManagerActivity.class));
                break;
            case R.id.button3:
                startActivity(new Intent(this, GridLayoutManager1Activity.class));
                break;
            case R.id.button4:
                startActivity(new Intent(this, GridLayoutManager2Activity.class));
                break;
            case R.id.button5:
                startActivity(new Intent(this, LinearLayoutManager2Activity.class));
                break;
            default:
                break;
        }
    }
}

File: sample/src/main/java/com/github/lany192/decoration/sample/MainActivity.kt
package com.github.lany192.decoration.sample

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.button -> startActivity(Intent(this, LinearLayoutManagerActivity::class.java))
            R.id.button2 -> startActivity(Intent(this, GridLayoutManagerActivity::class.java))
            R.id.button3 -> startActivity(Intent(this, GridLayoutManager1Activity::class.java))
            R.id.button4 -> startActivity(Intent(this, GridLayoutManager2Activity::class.java))
            R.id.button5 -> startActivity(Intent(this, LinearLayoutManager2Activity::class.java))
        }
    }
}